src/ui/linux/TogglDesktop/mainwindowcontroller.cpp

```diff
@@ -549,3 +555,21 @@ void MainWindowController::runScript() {
         quitApp();
     }
 }
+
+void MainWindowController::commitDataRequest() {
+    TogglApi::instance->stopEntryOnShutdown();
+}
+
+void MainWindowController::stateChanged(Qt::ApplicationState state) {
```

Comment:
This won't work in Linux, separate app suspension is something that happens on mobile platforms (iOS, Android, probably Windows) but to detect system suspending in Linux, we'll have to resort to using the `login1` DBus API (provided by `systemd` which is on most Linux systems nowadays) - the `org.freedesktop.login1.Manager.PrepareForSleep` signal can be used for that - and eventually (while we're at it), we could also use the `org.freedesktop.login1.Manager.PrepareForShutdown` signal too.

Reply:
That' cool. Let me research about those signals 👍